src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1623,27 +1625,14 @@ public Status do_GET_Document(Database _db, String docID, String _attachmentName
                 String revID = getQuery("rev");  // often null
                 RevisionInternal rev = null;
                 if (isLocalDoc) {
+                    // query only -> not required synchronized
```

Comment:
This comment might not be correct if want to make calling to the database object thread-safe.